aws-api/src/main/java/com/amplifyframework/api/aws/auth/AuthRuleProcessor.java

```diff
@@ -0,0 +1,222 @@
+                        .variable(key, "String!", value)
+                        .build();
+            } catch (AmplifyException exception) {
+                throw new ApiException(
+                        "Failed to set owner field on AppSyncGraphQLRequest", exception,
+                        "See attached exception for details.");
+            }
+        }
+
+        return request;
+    }
+
+    private boolean isReadRestrictingOwner(AuthRule authRule) {
+        return AuthStrategy.OWNER.equals(authRule.getAuthStrategy())
+                && authRule.getOperationsOrDefault().contains(ModelOperation.READ);
+    }
+
+    private boolean isReadRestrictingStaticGroup(AuthRule authRule) {
+        return AuthStrategy.GROUPS.equals(authRule.getAuthStrategy())
+                && authRule.getGroups() != null && !authRule.getGroups().isEmpty()
+                && authRule.getOperationsOrDefault().contains(ModelOperation.READ);
+    }
+
+    private String getIdentityValue(String identityClaim, AuthorizationType authType) throws ApiException {
+        String identityValue = null;
+
+        try {
+            identityValue = CognitoJWTParser
+                    .getPayload(getAuthToken(authType))
+                    .getString(identityClaim);
+        } catch (JSONException | CognitoParameterInvalidException error) {
+            // Could not read identity value from the token...
+            // Exception will be thrown so do nothing for now
+        }
+
+        if (identityValue == null || identityValue.isEmpty()) {
+            throw new ApiException(
+                    "Attempted to subscribe to a model with owner based authorization without " + identityClaim + " " +
+                            "which was specified (or defaulted to) as the identity claim.",
+                    "If you did not specify a custom identityClaim in your schema, make sure you are logged in. If " +
+                            "you did, check that the value you specified in your schema is present in the access key."
+            );
+        }
+
+        return identityValue;
+    }
+
+    private ArrayList<String> getUserGroups(AuthorizationType authType) throws ApiException {
+        // Custom groups claim isn't supported yet.
+        if (!AuthorizationType.AMAZON_COGNITO_USER_POOLS.equals(authType)) {
+            throw new ApiException("Custom groups claim is not supported yet.",
+                    "Please use Amazon Cognito User Pools to authorize your API.");
+        }
+
+        ArrayList<String> groups = new ArrayList<>();
+        final String GROUPS_KEY = "cognito:groups";
+
+        try {
+            JSONObject accessToken = CognitoJWTParser.getPayload(getAuthToken(authType));
+
+            if (accessToken.has(GROUPS_KEY)) {
+                JSONArray jsonGroups = accessToken.getJSONArray(GROUPS_KEY);
+
+                for (int i = 0; i < jsonGroups.length(); i++) {
+                    groups.add(jsonGroups.getString(i));
```

Comment:
Impossible to find single letter variables. I suggest:
```suggestion
                for (int index = 0; index < jsonGroups.length(); index++) {
                    groups.add(jsonGroups.getString(index));
```